src/components/Tooltip/index.js

```diff
@@ -1,37 +1,239 @@
+ * @return {DOMRect | null} The chosen bounding box. null if we failed to find
+ * a matching one, which can happen if the user is moving the mouse quickly
+ * and the onHoverOver event actually fires outside the element bounding box.
+ */
+function chooseBoundingBoxWithSlop(bbs, clientX, clientY, slop) {
+    for (let i = 0; i < bbs.length; i++) {
+        const bb = bbs[i];
+        if (bb.x - slop <= clientX && bb.x + bb.width + slop >= clientX && bb.y - slop <= clientY && bb.y + bb.height + slop >= clientY) {
+            return bb;
+        }
+    }
+    return null;
+}
+
+/**
+ * Choose the correct bounding box for the tooltip to be positioned against.
+ * This handles the case where the target is wrapped across two lines, and
+ * so we need to find the correct part (the one that the user is hovering
+ * over) and show the tooltip there.
+ *
+ * @param {Element} target The DOM element being hovered over.
+ * @param {number} clientX The X position from the MouseEvent.
+ * @param {number} clientY The Y position from the MouseEvent.
+ * @return {DOMRect} The chosen bounding box.
+ */
+function chooseBoundingBox(target, clientX, clientY) {
+    const bbs = target.getClientRects();
+    if (bbs.length === 1) {
+        return bbs[0];
+    }
+    let bb = chooseBoundingBoxWithSlop(bbs, clientX, clientY, 0);
+    if (bb) {
+        return bb;
+    }
+    // Retry with a slop factor, in case the user is moving the mouse quickly.
+    bb = chooseBoundingBoxWithSlop(bbs, clientX, clientY, 5);
+    if (bb) {
+        return bb;
+    }
+    // Fall back to the full bounding box if we failed to find a matching one.
+    // This could only happen if the user is moving the mouse very quickly
+    // and they got it outside our slop above.
+    return target.getBoundingClientRect();
+}
+
+/**
+ * A component used to wrap an element intended for displaying a tooltip. The term "tooltip's target" refers to the
+ * wrapped element, which, upon hover, triggers the tooltip to be shown.
+ * @param {propTypes} props
+ * @returns {ReactNodeLike}
+ */
+function Tooltip(props) {
+    const {children, numberOfLines, maxWidth, text, renderTooltipContent, renderTooltipContentKey} = props;
+
+    const {preferredLocale} = useLocalize();
+    const {windowWidth} = useWindowDimensions();
+
+    // Is tooltip already rendered on the page's body? happens once.
+    const [isRendered, setIsRendered] = useState(false);
+    // Is the tooltip currently visible?
+    const [isVisible, setIsVisible] = useState(false);
+    // The distance between the left side of the wrapper view and the left side of the window
+    const [xOffset, setXOffset] = useState(0);
+    // The distance between the top of the wrapper view and the top of the window
+    const [yOffset, setYOffset] = useState(0);
+    // The width and height of the wrapper view
+    const [wrapperWidth, setWrapperWidth] = useState(0);
+    const [wrapperHeight, setWrapperHeight] = useState(0);
```

Comment:
The height and width could be joined in a object, so there's less calls to update since they always update together. The same for the offsets.

Reply:
This code here has been moved to base tooltip, which already has the logics there.